addons/website_slides/static/src/js/public/components/slide_quiz_finish_dialog/slide_quiz_finish_dialog.js

```diff
@@ -0,0 +1,147 @@
+        const from = this.props.rankProgress.previous_rank.karma;
+        const to = this.props.rankProgress.new_rank.karma;
+        const fromLeft = this.props.rankProgress.previous_rank.progress;
+        const toLeft = this.props.rankProgress.new_rank.progress;
+
+        const animate = () => {
+            const progress = (Date.now() - startTime) / duration;
+            if (progress >= 1) {
+                this.state.tooltipLeft = toLeft;
+                this.state.tooltipValue = to;
+            } else {
+                this.state.tooltipLeft = Math.round(fromLeft + (toLeft - fromLeft) * progress);
+                this.state.tooltipValue = Math.round(from + (to - from) * progress);
+                browser.requestAnimationFrame(animate);
+            }
+        };
+        browser.requestAnimationFrame(animate);
+
+        // progress bar
+        if (this.props.rankProgress.level_up) {
+            this.state.title = _t("Level up!");
+            browser.setTimeout(() => {
+                this.state.hidePreviousRankBounds = false;
+                browser.setTimeout(() => {
+                    this.state.progressBarTransition = true;
+                    this.state.progressBarWidth = this.props.rankProgress.new_rank.progress;
+                }, 100);
+            }, 800);
+        } else {
+            this.state.hidePreviousRankBounds = false;
+            this.state.progressBarWidth = this.props.rankProgress.new_rank.progress;
+        }
+    }
+}
+
+export class SlideQuizFinishDialog extends Component {
+    static components = { Dialog, SlideXPProgressBar };
+    static props = {
+        close: Function,
+        hasNext: Boolean,
+        onClickNext: Function,
+        quiz: Object,
+        userId: Number,
+    };
+    static template = "website_slides.SlideQuizFinishDialog";
+
+    setup() {
+        super.setup();
+        this.state = useState({
+            animateKarmaGain: false,
+            fadeRankMotivational: false,
+            hideDismissBtns: true,
+            showRankMotivational: false,
+            title: _t("Amazing!"),
+        });
+        onMounted(() => {
+            this._animateText();
+        });
+    }
+
+    //--------------------------------
+    // Handlers
+    //--------------------------------
+
+    onClickNext() {
+        this.props.onClickNext();
+        this.props.close();
+    }
+
+    //--------------------------------
+    // Private
+    //--------------------------------
+
+    /**
+     * Handles the animation of the different text such as the karma gain
+     * and the motivational message when the user levels up.
+     * @private
+     */
+    _animateText() {
+        browser.setTimeout(() => {
+            this.state.animateKarmaGain = true;
+            this.state.hideDismissBtns = false;
+        }, 800);
+
+        if (this.props.quiz.rankProgress.level_up) {
+            this.state.title = _t("Level up!");
```

Comment:
Is the previous value really visible for a perceptible amount of time before we get here?
If not, we could have a "simple" `title` attribute instead of a state attribute.